test/e2e/broker_defaults_webhook_test.go

```diff
@@ -0,0 +1,178 @@
+// +build e2e
+
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	"knative.dev/eventing/pkg/apis/config"
+	eventingduck "knative.dev/eventing/pkg/apis/duck/v1"
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+	testlib "knative.dev/eventing/test/lib"
+	"knative.dev/pkg/system"
+)
+
+func TestBrokerNamespaceDefaulting(t *testing.T) {
+	const (
+		brokerClass = "MTChannelBasedBroker"
+	)
+	ctx := context.Background()
+
+	c := testlib.Setup(t, true)
+	defer testlib.TearDown(c)
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I know there's existing code that uses this, but it might be better to use the RetryTestErrors from github.com/knative/pkg/reconciler/retry.go
to reduce flakiness that we've been seeing.

Reply:
Nice catch. Done